File: internal/models/user_profile.go
package models

import "time"

// UserProfile represents a user profile for multi-profile premium users
type UserProfile struct {
	ID          int64     `json:"id"`
	TelegramID  int64     `json:"telegram_id"`
	PlayerTag   string    `json:"player_tag"`
	ProfileName *string   `json:"profile_name,omitempty"`
	IsPrimary   bool      `json:"is_primary"`
	CreatedAt   time.Time `json:"created_at"`
}

// NewUserProfile creates a new UserProfile instance
func NewUserProfile(telegramID int64, playerTag string) *UserProfile {
	return &UserProfile{
		TelegramID: telegramID,
		PlayerTag:  playerTag,
		IsPrimary:  false,
		CreatedAt:  time.Now(),
	}
}

File: oldpy/models/user_profile.py
"""
Модель профиля пользователя для многопрофильной системы
"""
from dataclasses import dataclass
from datetime import datetime
from typing import Optional


@dataclass
class UserProfile:
    """Модель профиля пользователя"""
    telegram_id: int
    player_tag: str
    profile_name: Optional[str] = None
    is_primary: bool = False
    created_at: Optional[str] = None
    profile_id: Optional[int] = None
    
    def __init__(self, telegram_id: int, player_tag: str, profile_name: str = None, 
                 is_primary: bool = False, created_at: str = None, profile_id: int = None):
        self.telegram_id = telegram_id
        self.player_tag = player_tag
        self.profile_name = profile_name
        self.is_primary = is_primary
        self.created_at = created_at or datetime.now().isoformat()
        self.profile_id = profile_id